packages/assets-controllers/src/MultichainAssetsRatesController/MultiChainTokensAssetsController.test.ts

```diff
@@ -0,0 +1,378 @@
+  id: 'account4',
+  type: 'bip122:p2wpkh',
+  address: '0x789',
+  metadata: {
+    name: 'EVM Account',
+    importTime: 0,
+    keyring: { type: 'bip122' },
+  },
+  scopes: [],
+  options: {},
+  methods: [],
+};
+
+// A fake conversion rates response returned by the SnapController.
+const fakeAccountRates = {
+  conversionRates: {
+    'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/slip44:501': {
+      'swift:0/iso4217:USD': {
+        rate: '202.11',
+        conversionTime: 1738539923277,
+      },
+    },
+  },
+};
+
+const setupController = ({
+  config,
+  accountsAssets = [fakeNonEvmAccount, fakeEvmAccount, fakeEvmAccount2],
+}: {
+  config?: Partial<
+    ConstructorParameters<typeof MultiChainAssetsRatesController>[0]
+  >;
+  accountsAssets?: InternalAccount[];
+} = {}) => {
+  const messenger = new Messenger<AllowedActions, AllowedEvents>();
+
+  messenger.registerActionHandler(
+    'MultichainAssetsController:getState',
+    () => ({
+      accountsAssets: {
+        account1: ['solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/slip44:501'],
+        account2: ['solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/slip44:501'],
+        account3: ['solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/slip44:501'],
+      },
+      assetsMetadata: {
+        'solana:5eykt4UsFv8P8NJdTREpY1vzqKqZKvdp/slip44:501': {
+          name: 'Solana',
+          symbol: 'SOL',
+          fungible: true,
+          iconUrl: 'https://example.com/solana.png',
+          units: [{ symbol: 'SOL', name: 'Solana', decimals: 9 }],
+        },
+      },
+    }),
+  );
+
+  messenger.registerActionHandler(
+    'AccountsController:listMultichainAccounts',
+    () => accountsAssets,
+  );
+
+  messenger.registerActionHandler('CurrencyRateController:getState', () => ({
+    currencyRates: {},
+    currentCurrency: 'USD',
+  }));
+
+  const multiChainAssetsRatesControllerMessenger = messenger.getRestricted({
+    name: 'MultiChainAssetsRatesController',
+    allowedActions: [
+      'AccountsController:listMultichainAccounts',
+      'SnapController:handleRequest',
+      'CurrencyRateController:getState',
+      'MultichainAssetsController:getState',
+    ],
+    allowedEvents: [
+      'AccountsController:accountAdded',
+      'KeyringController:lock',
+      'KeyringController:unlock',
+      'CurrencyRateController:stateChange',
+      'MultichainAssetsController:stateChange',
+    ],
+  });
+
+  return {
+    controller: new MultiChainAssetsRatesController({
+      messenger: multiChainAssetsRatesControllerMessenger,
+      ...config,
+    }),
+    messenger,
+  };
+};
+
+describe('MultiChainAssetsRatesController', () => {
+  let clock: sinon.SinonFakeTimers;
+
+  const mockedDate = 1705760550000;
+
+  beforeEach(() => {
+    clock = useFakeTimers();
+    jest.spyOn(Date, 'now').mockReturnValue(mockedDate);
+  });
+
+  afterEach(() => {
+    clock.restore();
+    jest.restoreAllMocks();
+  });
+
+  it('should initialize with an empty conversionRates state', () => {
```

Comment:
Few nit cosmetic comments;
should initialize with an empty conversionRates state

```suggestion
  it('initializes with an empty conversionRates state', () => {
```